src/euphorie/client/browser/risk.py

```diff
@@ -447,8 +451,43 @@ def proceed_to_next(self, reply):
         )
         return self.request.response.redirect(url)
 
+    @property
+    @memoize
+    def solutions(self):
+        if not self.risk:
```

Comment:
Given getattr is used a lot through the rest of the code I would go for the more concise:
```
return getattr(self.risk, "_solutions", [])
```

Reply:
makes sense, thanks!